de_lon8/Kareem/Mini_Project_1/Week 1 simple.py

```diff
@@ -173,43 +176,80 @@
-    #             order_phone_number = input('Please enter the phone number: ')
-    #             order_status = 'PREPARING'
+            #Return to previous menu
+            if orders_menu_choice == 0:
+                break
+            
+            #Prints all orders
+            elif orders_menu_choice == 1:
+                print(orders)
+
+            #Adding a new order
+            elif orders_menu_choice == 2:
+                order_name = input('Please enter the customer name: ')
+                order_address = (input('Enter the address number: '))
+                order_phone_number = input('Please enter the phone number: ')
+                order_status = 'PREPARING'
                 
-    #             single_order = {'Name': order_name, 'Address' : order_address, 'Phone Number' : order_phone_number, 'Status' : order_status}
+                single_order = {'Name': order_name, 'Address' : order_address, 'Phone Number' : order_phone_number, 'Status' : order_status}
 
-    #             orders.append(single_order)
+                orders.append(single_order)
 
-    #         elif orders_menu_choice == 3:
-    #             y = json.loads(orders)
-    #             print(y)
+            #Updating existing orderstatus
+            elif orders_menu_choice == 3:
+                order_number = 0
+                for i in orders:
+                    print (order_number, ':', i)
+                    order_number = order_number + 1
 
-    #         elif orders_menu_choice == 4:
-    #             print('')
+                to_amend = int(input('Which status would you like to change?: '))
                 
-    #         elif orders_menu_choice == 5:
-    #             print('')
+                status_progression = ('PREPARING', 'READY')
+                progression = 0
+                for i in status_progression:
+                    print (progression, ':', status_progression[progression])
+                    progression = progression + 1
+
+                amended_order = int(input ('What would you like to change the status to?'))
 
+                orders[to_amend]['Status'] = status_progression[amended_order]
+                
+                print(orders)
+
+            # #This is to update the existing orders
+            # elif orders_menu_choice == 4:
+            #     order_number = 0
+            #     for i in orders:
+            #         print (order_number, ':', i)
+            #         order_number = order_number + 1
+                
+            #     orders_updates = ('Name', 'Address', 'Phone Number')
+            #     updated_order = int(input('Which one of the inputs would you like to change? '))
+                
+            #     for key, value in orders[updated_order]:
+            #         print (key, value)
+                
+            
+            #This is deleting an entire order from the orders menu
+            elif orders_menu_choice == 5:
```

Comment:
What happens when we try to delete an order when the order list is empty? Have you tried executing that path of logic? Given that deleting an order is part of the specification, it might be an important feature we need to handle. Moreover, could a unit test have helped here?

Reply:
Thanks I totally forgot about the unit testing.
I will use it to generate the code for here especially.